### DISCUSSION TITLE
Usage of action "set_install_mode"

### DISCUSSION THREAD

**Discussion by dk1305hb** [SukramJ/aiohomematic#1953] [General]
Does anybody use the action "homematicip_local.set_install_mode" (Anlernmodus) successfully? I am trying it the first time with a new HmIP device and it fails.

<img width="1181" alt="Bildschirmfoto 2024-12-30 um 19 38 14" src="https://github.com/user-attachments/assets/dccb8cfe-b422-46fd-8ff5-7e39c0b0534a" />

The CCU internal log "hmserver.log" logs:
`2024-12-30 19:38:04,154 de.eq3.cbcs.legacy.bidcos.rpc.internal.LegacyAPIWorker ERROR [vert.x-worker-thread-4] Can not handle request java.lang.IllegalArgumentException: java.lang.ClassCastException@1f704470 
de.eq3.cbcs.legacy.communication.rpc.RpcException: java.lang.IllegalArgumentException: java.lang.ClassCastException@1f704470
	at de.eq3.cbcs.legacy.communication.rpc.internal.rpc.AnnotationAwareRpcHandler.execute(AnnotationAwareRpcHandler.java:84) ~[HMIPServer.jar:?]
`

I also tried it with the address of the new device, did not work.

I can't find any posts concerning this. Maybe it is successful with HM without IP devices.
If it would be successful, I ask me the next question, how could a new device moved out of the CCU internal "inbox"...?
<img width="925" alt="Bildschirmfoto 2024-12-30 um 19 41 52" src="https://github.com/user-attachments/assets/6ed1f4eb-5e32-41d9-9b30-e2a7162eabfd" />

**Comment by Baxxy13** [comment]
Obwohl `setInstallMode` auch für HmIP dokumentiert ist funktioniert die Methode (nach meiner Erfahrung) mit HmIP nicht.
Und selbst wenn es funktionieren würde hätte man ja immer noch das "Problem" das Gerät aus dem Posteingang der Zentrale zu bekommen. 
Meine Meinung: Diese Aktion kann weg weil man das Anlernen von Geräten besser direkt über die Zentrale macht.

**Reply by SukramJ** [reply]
Ich muss zugeben, das ich das nie ausprobiert habe, und schon lange vorgenommen habe, die Aktion auszubauen.
Ist vielleicht jetzt ein guter Zeitpunkt.

**Reply by dk1305hb** [reply]
Ok, das ist schade. Mit der Funktion könnte man den Anlernvorgang jemanden besser zugänglich machen, ohne dass man Know How in der CCU WebUi haben müsste. Außerdem wäre es möglich, die CCU so nur noch als Vermittler in den Hintergrund zu stellen.

**Reply by Baxxy13** [reply]
Eine CCU/RM war nie dafür ausgelegt als reines Gateway zu fungieren denn sie ist ja selbst eine "Smart Home Zentrale".
Und die Integration ist nicht dafür ausgelegt alle Funktionen der Zentrale abzubilden. 
Und das mit dem "Know How" bezüglich Anlernen lasse ich nicht gelten. 😉 
Es gibt ein WebUI-Handbuch in dem der Anlernvorgang beschrieben ist. 
In früheren Bedienungsanleitungen der Geräte wurde auch das Anlernen an eine CCU behandelt, heutzutage leider nicht mehr.

**Reply by dk1305hb** [reply]
Das kann ich alles verstehen. Ich habe selbst über 10 Jahre damit mein Smart Home aufgebaut.
Ich weiß halt nicht, ob wir irgendwann in nicht allzu ferner Zukunft an dem Zeitpunkt ankommen, dass die WebUI in einem modernen Browser nicht mehr funktioniert. Dann wird schnell dieser Wunsch aufkommen.

Mit Zugänglich meinte ich z.B. ein Familienmitglied wäre ohne WebUi Kenntnis in der Lage ein neues Gerät anzulernen, in dem man ein simples Dashboard in HA dafür zur Verfügung stellt. Also jemand mit nur HA Kenntnissen wäre dann dazu in der Lage.

**Reply by Baxxy13** [reply]
Ich kann deine Argumente mehr oder weniger nachvollziehen.
Aber solange die CCU3 / RM nicht für tot erklärt wurde wird schon dafür gesorgt werden das die WebUI weiterhin funktioniert.

Hab jetzt noch etwas recherchiert. Es sollte möglich sein auch den HmIP-Install-Mode "von extern" zu aktivieren.
Das Problem ist aber das man ja für jedes Interface alle möglichen Anlernmodi (Internet / Lokal) implementieren müsste weil dann vielleicht der Eine Dies und der Andere Das nutzen möchte. Und das macht die Sache "von HA aus" dann auch wieder kompliziert.

Es bliebe auch das Problem mit "das Gerät aus dem Posteingang holen". Gibt's sicher auch ne Methode für die gefunden und implementiert werden müsste.

Viel Aufwand um etwas "nachzubauen" was in der WebUI "out-of-the-box" seit Jahren funktioniert. 
Da ist es einfacher ne Kurzanleitung (für die Familienmitglieder) zu verfassen und diese griffbereit irgendwo abzulegen.

**Reply by dk1305hb** [reply]
Vielen Dank, dass du dir nochmal Zeit genommen hast. Solange wir keine Not haben, ist das auf jeden Fall nur ein nice to have. Und ich verstehe, dass @SukramJ das nicht leicht einfach mal so umsetzen kann. Die nachfolgende Problematik mit dem Posteingang habe ich mich ja auch schon gefragt. Ich wünsche euch einen guten Jahreswechsel.

**Reply by pdmetze** [reply]
Ich bin der gleichen Meinung wie sukramJ und Baxxy. Etwas Anlernen mache ich dann doch lieber im Quellsystem, hier Homematic, selbst. Den Aufwand das zu implementieren kann man sich aktuell sparen und wenn es jetzt nicht richtig funktioniert kann es m.E. raus.
